### PR TITLE
feat: add sortChildrenWith

### LINKED ISSUE #377: add sorting children by order function

### PR DESCRIPTION
**Pull request by pyphilia**
close #377

### CODE REVIEW

File: src/utils/item.test.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { v4 } from 'uuid';
 import { describe, expect, it } from 'vitest';
 
+import { Item } from '..';
```
Comment (spaenleh) [inline src/utils/item.test.ts:4]: Use `DiscriminatedItem` instead ?

File: src/utils/item.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Item } from '@/index';
+import { DiscriminatedItem, Item } from '@/index';
```
Comment (spaenleh) [inline src/utils/item.ts:1]: Actually We should not use `Item` in this file.

File: src/utils/item.ts [hunk 3]
```diff
@@ -53,9 +53,36 @@ export const isChildOf = (path: string, parentPath: string) => {
...
+    }
+
+    if (idsOrder.indexOf(ndElem.id) >= 0) {
+      return 1;
+    }
+
+    return stElem.createdAt < ndElem.createdAt ? -1 : 1;
+  };
+
+  return children.toSorted(compareFn);
```
Comment (spaenleh) [inline src/utils/item.ts:87]: nice !

File: tsconfig.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
     "module": "node16",
     "outDir": "dist",
     "esModuleInterop": true,
+    "lib": ["esnext"],
```
Comment (spaenleh) [inline tsconfig.json:7]: What is this used for ? :curious:

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good, but I think it should be DiscriminatedItem everywhere.

File: tsconfig.json (hunk 4 above)
Comment (pyphilia) [reply tsconfig.json:7]: it's necessary to use `toSorted`!

**Pull request merged**